modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/impl/PageMemoryNoStoreImpl.java

```diff
@@ -170,18 +170,19 @@ public class PageMemoryNoStoreImpl implements PageMemory {
      * @param directMemoryProvider Memory allocator to use.
      * @param dataRegionCfg Data region configuration.
      * @param ioRegistry IO registry.
+     * @param pageSize Page size in bytes.
      */
     public PageMemoryNoStoreImpl(
             DirectMemoryProvider directMemoryProvider,
-            PageMemoryDataRegionConfiguration dataRegionCfg,
-            PageIoRegistry ioRegistry
+            VolatilePageMemoryDataRegionConfiguration dataRegionCfg,
+            PageIoRegistry ioRegistry,
+            // TODO: IGNITE-17017 Move to common config
+            int pageSize
     ) {
         this.directMemoryProvider = directMemoryProvider;
         this.ioRegistry = ioRegistry;
         this.trackAcquiredPages = false;
-        this.dataRegionCfg = dataRegionCfg.value();
-
-        int pageSize = this.dataRegionCfg.pageSize();
+        this.dataRegionCfg = (VolatilePageMemoryDataRegionView) dataRegionCfg.value();
```

Comment:
Our naming makes no sense. Both configuration and view are named "dataRegionCfg", how do we distinct them in code?

Reply:
Fix it